src/dotnet-roslyn-tools/DartTest/DartTest.cs

```diff
@@ -0,0 +1,246 @@
+            var repositoryParams = new Dictionary<string, RepositoryResourceParameters>
+            {
+                {
+                    "self", new RepositoryResourceParameters
+                    {
+                        RefName = $"refs/heads/{azureBranchName}",
+                        Version = sha
+                    }
+                }
+            };
+
+            var runPipelineParameters = new RunPipelineParameters
+            {
+                Resources = new RunResourcesParameters
+                {
+
+                },
+                TemplateParameters = new Dictionary<string, string> { { "prNumber", prNumber.ToString() }, { "sha", sha } }
+            };
+
+            await remoteConnections.DevDivConnection.TryRunPipelineAsync(product.DartLabPipelineName, repositoryParams, runPipelineParameters, logger).ConfigureAwait(false);
+        }
+        catch (Exception e)
+        {
+            logger.LogError(e, e.Message);
+            return -1;
+        }
+        finally
+        {
+            CleanupDirectory(targetDirectory, logger);
+        }
+
+        return 0;
+    }
+
+    private static async Task<string?> GetLatestShaFromPullRequestAsync(IProduct product, HttpClient gitHubClient, int prNumber, ILogger logger, CancellationToken cancellationToken)
+    {
+        try
+        {
+            var requestUri = $"/repos/dotnet/{product.Name.ToLower()}/pulls/{prNumber}";
+            var response = await gitHubClient.GetAsync(requestUri, cancellationToken).ConfigureAwait(false);
+
+            if (response.IsSuccessStatusCode)
+            {
+                var prData = await response.Content.ReadFromJsonAsync<JsonObject>(cancellationToken: cancellationToken).ConfigureAwait(false);
+                return prData?["head"]?["sha"]?.ToString();
+            }
+            else
+            {
+                logger.LogError($"Failed to retrieve PR data from GitHub. Status code: {response.StatusCode}");
+            }
+        }
+        catch (Exception e)
+        {
+            logger.LogError(e, "Exception while retrieving the latest SHA from GitHub.");
+        }
+
+        return null;
+    }
+
+
+    private static async Task PushPRToInternalAsync(IProduct product, int prNumber, string azureBranchName, ILogger logger, string sha, string targetDirectory, CancellationToken cancellationToken)
+    {
+        try
+        {
+            var initCommand = $"init";
+            await RunGitCommandAsync(initCommand, logger, targetDirectory, cancellationToken).ConfigureAwait(false); ;
+
+            var addGithubRemoteCommand = $"remote add {product.Name.ToLower()} {product.RepoHttpBaseUrl}.git";
+            await RunGitCommandAsync(addGithubRemoteCommand, logger, targetDirectory, cancellationToken).ConfigureAwait(false);
+
+            var addInternalRemoteCommand = $"remote add internal {product.InternalRepoBaseUrl}";
```

Comment:
The TypeScript product won't have an internal mirror, so I think we need to fallback to the RepoBaseUrl when InternalRepoBaseUrl is empty string.